common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCache.java

```diff
@@ -515,16 +592,20 @@ public boolean removeAccount(@NonNull final AccountRecord accountToRemove) {
 
         final String cacheKey = mCacheValueDelegate.generateCacheKey(accountToRemove);
 
-        boolean accountRemoved = false;
-        if (mSharedPreferencesFileManager.keySet().contains(cacheKey))
-        {
-            mSharedPreferencesFileManager.remove(cacheKey);
-            accountRemoved = true;
-        }
+        synchronized (mCacheLock) {
+            waitForInitialLoad();
+            boolean accountRemoved = false;
+            if (mSharedPreferencesFileManager.keySet().contains(cacheKey))
+            {
+                mSharedPreferencesFileManager.remove(cacheKey);
+                accountRemoved = true;
+            }
+            Logger.info(methodTag, "Account was removed? [" + accountRemoved + "]");
```

Comment:
Based on the code, I think this log line may say `false` even in the case where Account was actually not present in `mSharedPreferencesFileManager`. I'm not sure if this should mark as removed FALSE vs removal not actually possible.

Reply:
There's nothing new here, this just moved around a bit with the caching changes. With that said, I think this looks correct.  The only case where something isn't removed is when it isn't present.  So I don't think there's any ambiguity in using a boolean. All the callers here also handle the boolean and expect these conditions.